progress.py

```diff
@@ -0,0 +1,10 @@
+import sys
+
+
+def print_sample_units_progress(number_processed, total):
```

Comment:
as discussed this function is nice, but could do with comments or ideally variable naming to make it clear what it's doing and why